pandas/core/generic.py

```diff
@@ -3123,6 +3124,18 @@ def to_latex(
         if multirow is None:
             multirow = config.get_option("display.latex.multirow")
 
+        if caption:
```

Comment:
instead of doing this here, can you do it in the formatter itself (in the constructor is ok)

Reply:
Done.